source/agora/node/Ledger.d

```diff
@@ -680,7 +680,7 @@ public class Ledger
         size_t active_enrollments = enroll_man.getValidatorCount(expect_height);
 
         if (data.enrolls.length + active_enrollments < Enrollment.MinValidatorCount)
-            return "Enrollment: Insufficient number of active validators";
+            return "Enrollment: Insufficient number of active validators"; // If you change this, change all occurences
```

Comment:
Shouldn't we enter this function with some valid state instead of patching it here ?

Reply:
If you mean that we should check for total enrollments for the next block earlier than this, we could. But this function is also used to validate the incoming `ConsensusData`, so this check should still be there probably.